test/qtismtest/data/storage/xml/marshalling/Html5FigcaptionMarshallerTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace qtismtest\data\storage\xml\marshalling;
+
+use qtism\data\content\xhtml\html5\Figcaption;
+use qtism\data\storage\xml\marshalling\MarshallerNotFoundException;
+use qtism\data\storage\xml\marshalling\MarshallingException;
+
+class Html5FigcaptionMarshallerTest extends Html5ElementMarshallerTest
+{
+    /**
+     * @throws MarshallerNotFoundException
+     * @throws MarshallingException
+     */
+    public function testMarshallerDoesNotExistInQti21(): void
+    {
+        $this->assertHtml5MarshallingOnlyInQti22AndAbove(new Figcaption(), Figcaption::QTI_CLASS_NAME);
+    }
+
+    /**
+     * @throws MarshallerNotFoundException
+     * @throws MarshallingException
+     */
+    public function testMarshall22(): void
+    {
+        $id = 'id';
+        $class = 'testclass';
+
+        $expected = sprintf(
+            '<%1$s id="%2$s " class="%3$s">text content</%1$s>',
```

Comment:
you are expecting **text content** whereas not adding it to the object that you serializing

Reply:
Could you please point me to how to do it without recreating the recursive marshaler for html5 element?

Reply:
What means without recreating?

Reply:
no matter I found the answer in PR which you did

Reply:
thank you for the help